pgmpy/models/FactorGraph.py

```diff
@@ -134,11 +137,69 @@ def remove_factors(self, *factors):
         >>> G = FactorGraph()
         >>> G.add_nodes_from(['a', 'b', 'c'])
         >>> phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        >>> G.add_edge('a',phi1)
         >>> G.add_factors(phi1)
         >>> G.remove_factors(phi1)
         """
         for factor in factors:
             self.factors.remove(factor)
+            if factor in super(FactorGraph, self).nodes():
```

Comment:
@kris-singh I think we had decided on deleting the `remove_factor` method. @yashu-seth